### PR TITLE
new resource: `azurerm_sentinel_data_connector_aws_cloud_trail`

### PR DESCRIPTION
**Pull request by magodo**
new resource: `azurerm_sentinel_data_connector_aws_cloud_trail`

Note that since `azurerm_sentinel_data_connector_aws_cloud_trail` needs additional setup to do the test, I provide following execution result to prove it works:

<details open>
<summary>Test Result</summary>
<pre>
    $ tf apply -var arn="arn:aws:iam::xxxx:role/magodo-sentinel"
    azurerm_resource_group.test: Refreshing state... [id=/subscriptions/xxxxxxxx/resourceGroups/acctestRG-sentinel-20210121]
    azurerm_log_analytics_workspace.test: Refreshing state... [id=/subscriptions/xxxx/resourceGroups/acctestRG-sentinel-20210121/providers/Microsoft.OperationalInsights/workspaces/acctest-workspace-20210121]
    azurerm_log_analytics_solution.test: Refreshing state... [id=/subscriptions/xxxx/resourcegroups/acctestRG-sentinel-20210121/providers/Microsoft.OperationsManagement/solutions/SecurityInsights(acctest-workspace-20210121)]

    An execution plan has been generated and is shown below.
    Resource actions are indicated with the following symbols:
      + create

    Terraform will perform the following actions:

      # azurerm_sentinel_data_connector_aws_cloud_trail.test will be created
      + resource "azurerm_sentinel_data_connector_aws_cloud_trail" "test" {
          + aws_role_arn               = "arn:aws:iam::xxxx:role/magodo-sentinel"
          + id                         = (known after apply)
          + log_analytics_workspace_id = "/subscriptions/xxxx/resourceGroups/acctestRG-sentinel-20210121/providers/Microsoft.OperationalInsights/workspaces/acctest-workspace-20210121"
          + name                       = "accTestDC-123"
          + tenant_id                  = (known after apply)
        }

    Plan: 1 to add, 0 to change, 0 to destroy.

    Do you want to perform these actions?
      Terraform will perform the actions described above.
      Only 'yes' will be accepted to approve.

      Enter a value: yes

    azurerm_sentinel_data_connector_aws_cloud_trail.test: Creating...
    azurerm_sentinel_data_connector_aws_cloud_trail.test: Creation complete after 6s [id=/subscriptions/xxxx/resourceGroups/acctestRG-sentinel-20210121/providers/Microsoft.OperationalInsights/workspaces/acctest-workspace-20210

    Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

    github/tf-config/sentinel_dc_sdk via M-p~_~RM-  default took 37s
    M-p~_~RM-$ tf apply -var arn="arn:aws:iam::xxxx:role/magodo-sentinel2"
    azurerm_resource_group.test: Refreshing state... [id=/subscriptions/xxxx/resourceGroups/acctestRG-sentinel-20210121]
    azurerm_log_analytics_workspace.test: Refreshing state... [id=/subscriptions/xxxx/resourceGroups/acctestRG-sentinel-20210121/providers/Microsoft.OperationalInsights/workspaces/acctest-workspace-20210121]
    azurerm_sentinel_data_connector_aws_cloud_trail.test: Refreshing state... [id=/subscriptions/xxxx/resourceGroups/acctestRG-sentinel-20210121/providers/Microsoft.OperationalInsights/workspaces/acctest-workspace-20210121/pro    azurerm_log_analytics_solution.test: Refreshing state... [id=/subscriptions/xxxx/resourcegroups/acctestRG-sentinel-20210121/providers/Microsoft.OperationsManagement/solutions/SecurityInsights(acctest-workspace-20210121)]

    An execution plan has been generated and is shown below.
    Resource actions are indicated with the following symbols:
      ~ update in-place

    Terraform will perform the following actions:

      # azurerm_sentinel_data_connector_aws_cloud_trail.test will be updated in-place
      ~ resource "azurerm_sentinel_data_connector_aws_cloud_trail" "test" {
          ~ aws_role_arn               = "arn:aws:iam::xxxx:role/magodo-sentinel" -> "arn:aws:iam::xxxx:role/magodo-sentinel2"
            id                         = "/subscriptions/xxxx/resourceGroups/acctestRG-sentinel-20210121/providers/Microsoft.OperationalInsights/workspaces/acctest-workspace-20210121/providers/Microsoft.SecurityInsights/dataCo            name                       = "accTestDC-123"
            # (1 unchanged attribute hidden)
        }

    Plan: 0 to add, 1 to change, 0 to destroy.

    Do you want to perform these actions?
      Terraform will perform the actions described above.
      Only 'yes' will be accepted to approve.

      Enter a value: yes

    azurerm_sentinel_data_connector_aws_cloud_trail.test: Modifying... [id=/subscriptions/xxxx/resourceGroups/acctestRG-sentinel-20210121/providers/Microsoft.OperationalInsights/workspaces/acctest-workspace-20210121/providers/    azurerm_sentinel_data_connector_aws_cloud_trail.test: Modifications complete after 5s [id=/subscriptions/xxxx/resourceGroups/acctestRG-sentinel-20210121/providers/Microsoft.OperationalInsights/workspaces/acctest-workspace-

    Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
</pre>
</details>

### CODE REVIEW

**Comment by katbyte** [comment]
@magodo - build is failing: 
```
38]	[Step 4/5] # github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/sentinel
[20:49:38]	[Step 4/5] ./sentinel_data_connector_aws_cloud_trail.go:84:9: undefined: dataConnectorID
```

**Comment by magodo** [comment]
@katbyte I've fix the build error and some other minor changes commented on other DC PRs. Please take another review.

File: azurerm/internal/services/sentinel/sentinel_data_connector_aws_cloud_trail.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	if !d.IsNewResource() {
+		resp, err := client.Get(ctx, id.ResourceGroup, operationalInsightsResourceProvider, id.WorkspaceName, name)
+		if err != nil {
+			return fmt.Errorf("retrieving %s: %+v", id, err)
+		}
+
+		if err := assertDataConnectorKind(resp.Value, securityinsight.DataConnectorKindAmazonWebServicesCloudTrail); err != nil {
+			return fmt.Errorf("asserting %s: %+v", id, err)
+		}
+		param.Etag = resp.Value.(securityinsight.AwsCloudTrailDataConnector).Etag
```
Comment (katbyte) [inline azurerm/internal/services/sentinel/sentinel_data_connector_aws_cloud_trail.go:112]: question still is unresolved in 365 data

File: azurerm/internal/services/sentinel/sentinel_data_connector_aws_cloud_trail.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	}
+
+	_, err = client.CreateOrUpdate(ctx, id.ResourceGroup, operationalInsightsResourceProvider, id.WorkspaceName, id.Name, param)
+	if err != nil {
```
Comment (katbyte) [inline azurerm/internal/services/sentinel/sentinel_data_connector_aws_cloud_trail.go:116]: these lines could be meged

File: azurerm/internal/services/sentinel/sentinel_data_connector_aws_cloud_trail.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.DataConnectorID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	_, err = client.Delete(ctx, id.ResourceGroup, operationalInsightsResourceProvider, id.WorkspaceName, id.Name)
+	if err != nil {
```
Comment (katbyte) [inline azurerm/internal/services/sentinel/sentinel_data_connector_aws_cloud_trail.go:172]: these lines could be merged

**Comment by magodo** [comment]
The etag related code can be removed once: https://github.com/Azure/azure-rest-api-specs/issues/13203 is resolved.

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Pull request merged**